src/API/Site/Controllers/MerchantCenter/SyncableProductsCountController.php

```diff
@@ -0,0 +1,141 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\MerchantCenter;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\BaseController;
+use Automattic\WooCommerce\GoogleListingsAndAds\API\TransportMethods;
+use Automattic\WooCommerce\GoogleListingsAndAds\Jobs\JobRepository;
+use Automattic\WooCommerce\GoogleListingsAndAds\Jobs\UpdateSyncableProductsCount;
+use Automattic\WooCommerce\GoogleListingsAndAds\Options\TransientsInterface;
+use Automattic\WooCommerce\GoogleListingsAndAds\Proxies\RESTServer;
+use WP_REST_Request as Request;
+use WP_REST_Response as Response;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class SyncableProductsCountController
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\MerchantCenter
+ */
+class SyncableProductsCountController extends BaseController {
+
+	/**
+	 * @var TransientsInterface
+	 */
+	protected $transients;
+
+	/**
+	 * @var JobRepository
+	 */
+	protected $job_repository;
+
+	/**
+	 * SyncableProductsCountController constructor.
+	 *
+	 * @param RESTServer          $server
+	 * @param TransientsInterface $transients
+	 * @param JobRepository       $job_repository
+	 */
+	public function __construct( RESTServer $server, TransientsInterface $transients, JobRepository $job_repository ) {
+		parent::__construct( $server );
+		$this->transients     = $transients;
+		$this->job_repository = $job_repository;
+	}
+
+
+	/**
+	 * Registers the routes for the objects of the controller.
+	 */
+	public function register_routes() {
+		$this->register_route(
+			'mc/syncable-products-count',
+			[
+				[
+					'methods'             => TransportMethods::READABLE,
+					'callback'            => $this->get_syncable_products_count_callback(),
+					'permission_callback' => $this->get_permission_callback(),
+				],
+			]
+		);
+
+		$this->register_route(
+			'mc/syncable-products-count',
+			[
+				[
+					'methods'             => TransportMethods::CREATABLE,
+					'callback'            => $this->update_syncable_products_count_callback(),
+					'permission_callback' => $this->get_permission_callback(),
+				],
+			]
+		);
+	}
+
+	/**
+	 * Get the callback function for marking setup complete.
+	 *
+	 * @return callable
+	 */
+	protected function get_syncable_products_count_callback(): callable {
+		return function( Request $request ) {
+			$response = [
+				'count' => null,
+			];
+
+			$count = (int) $this->transients->get( TransientsInterface::SYNCABLE_PRODUCTS_COUNT );
+
+			if ( $count ) {
+				$response['count'] = $count;
+			}
+
+			return $this->prepare_item_for_response( $response, $request );
+		};
+	}
+
+	/**
+	 * Get the callback for syncing shipping.
+	 *
+	 * @return callable
+	 */
+	protected function update_syncable_products_count_callback(): callable {
+		return function( Request $request ) {
+			$job = $this->job_repository->get( UpdateSyncableProductsCount::class );
+			$job->schedule();
```

Comment:
Should we be clearing the transient, when we trigger this job? Otherwise it might get a stale count thinking it's already completed if we call a GET request right after.

Reply:
According to the below comment https://github.com/woocommerce/google-listings-and-ads/pull/1699#discussion_r980712187 I'm going to change to use an option rather than a transient. I will still add a line here to remove the option before we run the job so the frontend would not get a stale count.

Reply:
Updated in 25f3ee3 and 8a48750.